### ISSUE TITLE
Jetpack Plans: do not offer CNAME prompt

### ISSUE THREAD

**Issue by kraftbj** [Automattic/wp-calypso#8903]
With a premium plan for a Jetpack site on a subdomain, I was offered the alert that my CNAME records needed to be updated. I assume this was intended for dotcom sites with a mapped subdomain when the CNAME didn't point to the original wp.com site.
![keep1](https://cloud.githubusercontent.com/assets/88897/19599721/b56c9516-9767-11e6-91d7-9b0e7ffe1910.png)

**Comment by kraftbj** [comment]
This is also impacting root domains being asked to update their A records.

**Comment by rickybanister** [comment]
cc @klimeryk—it looks like you set up these domain warnings.

**Comment by klimeryk** [comment]
Indeed - I did not notice that `DomainWarnings` have been added to another view, `CurrentPlan`. @lamosty, could I ask for your assistance with this file? It's heavily reduxified and I honestly don't know what would be the most "proper" way of short-circuiting fetching the domains if the site is a Jetpack one 😊
For reference, this is a PR where I've fixed a similar (but simpler) case: https://github.com/Automattic/wp-calypso/pull/8585
What I wanted to achieve with that PR is to _not_ fetch the domains at all from the backend - because they are useless on a Jetpack site and we are wasting time and resources on a request to the backend for them. I think the same should be done in https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/plans/current-plan/index.jsx too.
